applications/serial_lte_modem/src/slm_at_commands.c

```diff
@@ -311,258 +326,146 @@ static int handle_at_datactrl(enum at_cmd_type cmd_type)
 
-int handle_at_clac(enum at_cmd_type cmd_type);
-
-/* TCP proxy commands */
-int handle_at_tcp_server(enum at_cmd_type cmd_type);
-int handle_at_tcp_client(enum at_cmd_type cmd_type);
-int handle_at_tcp_send(enum at_cmd_type cmd_type);
-int handle_at_tcp_hangup(enum at_cmd_type cmd_type);
-
-/* UDP proxy commands */
-int handle_at_udp_server(enum at_cmd_type cmd_type);
-int handle_at_udp_client(enum at_cmd_type cmd_type);
-int handle_at_udp_send(enum at_cmd_type cmd_type);
-
-/* Socket-type TCPIP commands */
-int handle_at_socket(enum at_cmd_type cmd_type);
-int handle_at_secure_socket(enum at_cmd_type cmd_type);
-int handle_at_socket_select(enum at_cmd_type cmd_type);
-int handle_at_socketopt(enum at_cmd_type cmd_type);
-int handle_at_secure_socketopt(enum at_cmd_type cmd_type);
-int handle_at_bind(enum at_cmd_type cmd_type);
-int handle_at_connect(enum at_cmd_type cmd_type);
-int handle_at_listen(enum at_cmd_type cmd_type);
-int handle_at_accept(enum at_cmd_type cmd_type);
-int handle_at_send(enum at_cmd_type cmd_type);
-int handle_at_recv(enum at_cmd_type cmd_type);
-int handle_at_sendto(enum at_cmd_type cmd_type);
-int handle_at_recvfrom(enum at_cmd_type cmd_type);
-int handle_at_poll(enum at_cmd_type cmd_type);
-int handle_at_getaddrinfo(enum at_cmd_type cmd_type);
-
-#if defined(CONFIG_SLM_NATIVE_TLS)
-int handle_at_xcmng(enum at_cmd_type cmd_type);
-#endif
-
-/* ICMP commands */
-int handle_at_icmp_ping(enum at_cmd_type cmd_type);
 
-#if defined(CONFIG_SLM_SMS)
-/* SMS commands */
-int handle_at_sms(enum at_cmd_type cmd_type);
-#endif
-
-/* FOTA commands */
-int handle_at_fota(enum at_cmd_type cmd_type);
-
-#if defined(CONFIG_SLM_GNSS)
-int handle_at_gps(enum at_cmd_type cmd_type);
-int handle_at_gps_delete(enum at_cmd_type cmd_type);
-#endif
-
-#if defined(CONFIG_SLM_FTPC)
-int handle_at_ftp(enum at_cmd_type cmd_type);
-#endif
-#if defined(CONFIG_SLM_TFTPC)
-int handle_at_tftp(enum at_cmd_type cmd_type);
-#endif
-
-#if defined(CONFIG_SLM_MQTTC)
-int handle_at_mqtt_config(enum at_cmd_type cmd_type);
-int handle_at_mqtt_connect(enum at_cmd_type cmd_type);
-int handle_at_mqtt_publish(enum at_cmd_type cmd_type);
-int handle_at_mqtt_subscribe(enum at_cmd_type cmd_type);
-int handle_at_mqtt_unsubscribe(enum at_cmd_type cmd_type);
-#endif
-
-#if defined(CONFIG_SLM_HTTPC)
-int handle_at_httpc_connect(enum at_cmd_type cmd_type);
-int handle_at_httpc_request(enum at_cmd_type cmd_type);
-#endif
-
-#if defined(CONFIG_SLM_TWI)
-int handle_at_twi_list(enum at_cmd_type cmd_type);
-int handle_at_twi_write(enum at_cmd_type cmd_type);
-int handle_at_twi_read(enum at_cmd_type cmd_type);
-int handle_at_twi_write_read(enum at_cmd_type cmd_type);
-#endif
-
-#if defined(CONFIG_SLM_GPIO)
-int handle_at_gpio_configure(enum at_cmd_type cmd_type);
-int handle_at_gpio_operate(enum at_cmd_type cmd_type);
-#endif
-
-static struct slm_at_cmd {
-	const char *string;
-	slm_at_handler_t handler;
-} slm_at_cmd_list[] = {
+static const char * const slm_at_cmd_list[] = {
```

Comment:
Haven't tested, and it might be a bit hack-ish, though perhaps you can do something like 

```
size_t cmd_custom_len = _nrf_modem_at_cmd_custom_list_end - _nrf_modem_at_cmd_custom_list_start;
char * slm_at_cmd_list[cmd_custom_len];

extern struct nrf_modem_at_cmd_custom _nrf_modem_at_cmd_custom_list_start[];

for (int i = 0; i < cmd_custom_len  ; i++) {
        slm_at_cmd_list[i] = _nrf_modem_at_cmd_custom_list_start[i].cmd;
}
```

Reply:
> Note that this will include if there are other libraries etc. that are using custom commands as well, and for commands that exist in the modem that there is a wrapper for.

Ah, true. This would need filtering to only pick the commands starting with `AT#` so that only the SLM-specific commands are picked up.